keepassxc-browser/content/ui.js

```diff
@@ -144,12 +144,15 @@ kpxcUI.isRTL = function(field) {
         return false;
     }
 
-    if (document.dir.toLowerCase() === 'rtl' || document.body.getLowerCaseAttribute('dir')) {
+    const style = getComputedStyle(field);
+    const fieldDir = field.getLowerCaseAttribute('dir');
+    if (style.textAlign.toLowerCase() === 'left' || fieldDir === 'ltr') {
```

Comment:
if textalign is set to *either* right or left, the value of `dir` makes no difference.

For example, consider `textalign=right` and d `fieldDir=='ltr', then this first conditional would match
and isRTL would be set to false. But if `textalign=right` the text will be right aligned so this won't work.

Reply:
That's the priority? `text-align` always first, then the field attribute.

Reply:
yes, I toggled things to see what takes precedence and text-align always wins, then the value of the dir property starting with the element itself if present or else inheriting from any ancestor that has `dir` attribute set, going up the ancestors until the topmost `<html>` tag.

Reply:
How's this?
```js
kpxcUI.isRTL = function(elem) {
    if (!elem) {
        return false;
    }

    const style = getComputedStyle(elem);
    if (style.textAlign.toLowerCase() === 'left') {return false;}
    else if (style.textAlign.toLowerCase() === 'right') {return true;}

    const loopLimit = 128;
    for (let i = 0; i < loopLimit; i++) {
        const elemDir = elem.getLowerCaseAttribute('dir');
        if (elemDir === 'ltr') { return false; }
        else if (elemDir === 'rtl') { return true; }
        elem = elem.parentElement;
        if (!elem) break;
    }

    return false;
};

```

Reply:
We already have a `kpxcFields.traverseParents()` function for these kind of situations.

Reply:
Like this?
```js
kpxcUI.isRTL = function(element) {
    if (!element) {
        return false;
    }

    const style = getComputedStyle(element);
    if (style.textAlign.toLowerCase() === 'left') {return false;}
    else if (style.textAlign.toLowerCase() === 'right') {return true;}

    return kpxcFields.traverseParents(element, 
        (element) => ["ltr", "rtl"].includes(element.getLowerCaseAttribute('dir')),
        (element) => ({"ltr" : false, "rtl" : true})[element.getLowerCaseAttribute('dir')]
    ) 
};
```

Reply:
Something like that yes :)

Reply:
steal my code... *please*!

Reply:
Works great. Thanks!

Reply:
lgtm